main.js

```diff
@@ -9,45 +8,46 @@ var bearBox = document.getElementById("bear-container");
 var outfits =[];
 window.localStorage.setItem('id','0');
 var currentOutfit = new Outfit();
-var garmentNodeList = generateNestedNodeList();
 
 window.localStorage.setItem('currentBackground', 'blue')
 
 backgroundButtons.addEventListener('click', changeBackground);
 itemButtonParent.addEventListener('click', changeGarment);
 saveButton.addEventListener('click', saveOutfit);
 
-function generateNestedNodeList(){
-  var tempNodeList = Array.prototype.slice.call(document.querySelectorAll('.item-button'));
-  return[tempNodeList.slice(0,4),tempNodeList.slice(4,7),tempNodeList.slice(7,12)]
-}
 
 function changeBackground() {
   if (event.target.classList.contains('item-button')) {
     var background = event.target.innerText.toLowerCase();
     var currentBackground = window.localStorage.getItem('currentBackground');
     window.localStorage.setItem('currentBackground', background)
-    bearBox.classList.remove(currentBackground);
+    currentOutfit.background = currentBackground;
+    bearBox.classList.remove(background);
     bearBox.classList.add(background);
   }
 }
+
 function changeGarment() {
-  var garmentName = event.target.innerText;
-  for (var i = 0; i < garmentNodeList.length; i++) {
-    for (var j = 0; j < garmentNodeList[i].length; j++) {
-      if (garmentNodeList[i][j].innerText === garmentName) {
-        garmentName = garmentName.toLowerCase().replace(/ /g,'');
-        currentOutfit.addGarment(i, garmentName);
-      }
-    }
+  switch (event.target.parentElement.id) {
+    case 'hat-buttons':
+      currentOutfit.addGarment(0, event.target.innerText)
+      break;
+    case 'clothes-buttons':
+      currentOutfit.addGarment(1, event.target.innerText)
+      break;
+    case 'accessory-buttons':
+      currentOutfit.addGarment(2, event.target.innerText)
+      break;
```

Comment:
Nice, I didn't know about switch statements, but this is a much better solution!